boxoffice/views/admin_discount.py

```diff
@@ -0,0 +1,197 @@
+from utils import xhr_only, date_time_format, api_error, api_success
+
+
+def jsonify_price(price):
+    if price:
+        return {
+            'price_title': price.title,
+            'amount': price.amount,
+            'start_at': date_time_format(price.start_at),
+            'end_at': date_time_format(price.end_at)
+        }
+    else:
+        return None
+
+
+def jsonify_discount_policy(policy):
+    return {
+        'id': policy.id,
+        'name': policy.name,
+        'title': policy.title,
+        'discount_type': "Automatic" if policy.is_automatic else "Coupon based",
+        'item_quantity_min': policy.item_quantity_min,
+        'percentage': policy.percentage,
+        'is_price_based': policy.is_price_based,
+        'discount_code_base': policy.discount_code_base,
+        'bulk_coupon_usage_limit': policy.bulk_coupon_usage_limit,
+        'discount': policy.percentage if not policy.is_price_based else '',
+        'price_details': jsonify_price(Price.query.filter(Price.discount_policy == policy).first()) if policy.is_price_based else '',
+        'currency': CURRENCY_SYMBOL['INR'],
+        'dp_items': [{'id': str(item.id), 'title': item.title} for item in policy.items]
+    }
+
+
+def jsonify_discount_policies(data_dict):
+    discount_policies_list = []
+    for policy in data_dict['discount_policies']:
+        discount_policies_list.append(jsonify_discount_policy(policy))
+    return jsonify(
+        org_name=data_dict['org'].name, title=data_dict['org'].title,
+        discount_policies=discount_policies_list,
+        total_pages=data_dict['total_pages'],
+        paginated=data_dict['total_pages'] > 1,
+        current_page=data_dict['current_page']
+    )
+
+
+@app.route('/api/1/admin/<org>/discount_policy')
+@lastuser.requires_login
+@load_models(
+    (Organization, {'name': 'org'}, 'organization'),
+    permission='org_admin'
+    )
+@render_with({'text/html': 'index.html', 'application/json': jsonify_discount_policies}, json=True)
+@requestargs('search', ('page', int))
+def admin_discount_policies(organization, search=None, page=1):
+    results_per_page = 20
+
+    if request.is_xhr:
+        discount_policies = organization.discount_policies
+
+        if search:
+            discount_policies = discount_policies.filter(
+                DiscountPolicy.title.ilike('%{query}%'.format(query=search)))
+
+        total_policies = discount_policies.count()
+        total_pages = int(math.ceil(total_policies / results_per_page))
+        offset = (page - 1) * results_per_page
+
+        discount_policies = discount_policies.limit(results_per_page).offset(offset).all()
+
+        return dict(
+            org=organization, title=organization.title,
+            discount_policies=discount_policies,
+            total_pages=total_pages,
+            paginated=(total_policies > results_per_page),
+            current_page=page)
+    else:
+        return dict(org=organization, title=organization.title)
```

Comment:
Since `jsonify_discount_policies` only seems to need `organization.name` and `organization.title`, maybe just return those two attributes only, instead of returning the entire object?